### PR TITLE
BUG: fix interpolation not writing to CUDA arrays

### LINKED ISSUE #347: Interpolation fails silently for cuda spaces
The following line fails to write to the output in case of cuda data.

```python
            func(mesh, out=out.asarray().reshape(self.grid.shape,
                                                 order=self.order))
```

https://github.com/odlgroup/odl/blob/master/odl/discr/discr_mappings.py#L273-L274

### PR DESCRIPTION
**Pull request by kohr-h**
Should fix #347

### CODE REVIEW

**Comment by adler-j** [comment]
Perhaps we should have a test as well?

**Comment by kohr-h** [comment]
Here comes the test, works on my machine.

File: test/discr/discr_mappings_test.py [hunk 1]
```diff
@@ -345,5 +346,19 @@ def test_collocation_interpolation_identity():
...
 
 
+@skip_if_no_cuda
+def test_collocation_cuda():
+    rect = odl.Rectangle([0, 0], [1, 1])
+    part = odl.uniform_partition_fromintv(rect, [4, 2])
+    space = odl.FunctionSpace(rect)
+    dspace = odl.CudaRn(part.size)
+
+    coll_op = PointCollocation(space, part, dspace, order='F')
```
Comment (adler-j) [inline test/discr/discr_mappings_test.py]: Why order='F'?

**Comment by adler-j** [comment]
Minor note, otherwise looks fine.

File: test/discr/discr_mappings_test.py (hunk 1 above)
Comment (kohr-h) [inline test/discr/discr_mappings_test.py]: Hm, coincidence I guess. Wasn't Fortran ordering the natural thing on CUDA spaces?

Comment (adler-j) [inline test/discr/discr_mappings_test.py]: Yes, but I think we should stay with the defaults until such a time that we change that.

**Comment by kohr-h** [comment]
Found another CUDA related bug along the way, triggered by the new test. Merging after Travis is happy.

**Comment by kohr-h** [comment]
And yet another bug. Now we ravel before assignment.

**Pull request merged**